tfhe/src/high_level_api/tests/noise_distribution/encryption_noise.rs

```diff
@@ -0,0 +1,227 @@
+        V0_11_PARAM_KEYSWITCH_PKE_TO_BIG_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64_ZKV1,
+        PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64,
+    )
+}
+
+fn noise_check_compact_public_key_encryption_noise_tuniform(
+    mut cpke_params: CompactPublicKeyEncryptionParameters,
+    ksk_params: ShortintKeySwitchingParameters,
+    block_params: ClassicPBSParameters,
+) {
+    // Hack to avoid server key needs and get the ciphertext directly
+    cpke_params.expansion_kind =
+        CompactCiphertextListExpansionKind::NoCasting(block_params.encryption_key_choice.into());
+
+    let modulus_as_f64 = if cpke_params.ciphertext_modulus.is_native_modulus() {
+        2.0f64.powi(u64::BITS as i32)
+    } else {
+        cpke_params.ciphertext_modulus.get_custom_modulus() as f64
+    };
+
+    let encryption_noise = cpke_params.encryption_noise_distribution;
+    let encryption_variance = match encryption_noise {
+        DynamicDistribution::Gaussian(_gaussian) => {
+            panic!("This test is written for TUniform, wrong parameter set used")
+        }
+        DynamicDistribution::TUniform(tuniform) => tuniform.variance(modulus_as_f64),
+    };
+
+    let expected_variance = lwe_compact_public_key_encryption_expected_variance(
+        encryption_variance,
+        cpke_params.encryption_lwe_dimension,
+    );
+
+    let config = ConfigBuilder::with_custom_parameters(block_params)
+        .use_dedicated_compact_public_key_parameters((cpke_params, ksk_params))
+        .build();
+
+    const NB_TEST: usize = 16000;
+
+    let mut noise_samples = Vec::with_capacity(NB_TEST);
+    for _ in 0..NB_TEST {
+        let cks = ClientKey::generate(config);
+        let cpk = CompactPublicKey::new(&cks);
+
+        let mut builder = CompactCiphertextList::builder(&cpk);
+        builder
+            .push_with_num_bits(0u32, cpke_params.message_modulus.0.ilog2() as usize)
+            .unwrap();
+        let list = builder.build();
+        let expanded = list.expand().unwrap();
+        let encrypted: FheUint2 = expanded.get(0).unwrap().unwrap();
+
+        // Drop to lower level APIs to get to the noise
+        let integer_key = cks.into_raw_parts().1.unwrap().0;
+        let shortint_key = integer_key.into_raw_parts();
+        let core_key = shortint_key.into_raw_parts().0;
+
+        let radix = encrypted.into_raw_parts().0;
+        assert!(radix.blocks.len() == 1);
+        let shortint_block = radix.blocks.into_iter().next().unwrap();
+        let lwe_ct = shortint_block.ct;
+
+        // This is directly the noise as we encrypted a 0
+        let noise = decrypt_lwe_ciphertext(&core_key, &lwe_ct).0;
+        let signed_noise: i64 = noise as i64;
+
+        // Rescale
+        noise_samples.push(signed_noise as f64 / modulus_as_f64);
+    }
+
+    let measured_variance = variance(&noise_samples);
+    let measured_confidence_interval =
+        variance_confidence_interval(noise_samples.len() as f64, measured_variance, 0.99);
```

Comment:
`confidence_level` variable could be defined above and used here.

A significance level of 1% produces a quite flaky test.

Reply:
haven't had issues when we tested those up until now

Reply:
Sure, it's not that big
But 1% can be quite annoying, it means 1% of the CI runs will fail because of this test I believe

Reply:
I'll run in a loop and try to get an estimate of failures

Reply:
0 fails in a 1000 run

Reply:
Then either the confidence interval is not correct or I don't interpret the confidence level well
What's your understanding of what the confidence level is?